WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Notes Section/Customer Note/EditCustomerNoteViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import Yosemite
+import Combine
+
+/// View Model for the Edit Customer Note screen
+///
+final class EditCustomerNoteViewModel: ObservableObject {
+
+    /// New content to submit.
+    /// Binding property modified at the view level.
+    ///
+    @Published var newNote: String
+
+    /// Active navigation bar trailing item.
+    /// Defaults to a disabled done button.
+    ///
+    @Published private(set) var navigationTrailingItem: NavigationItem = .done(enabled: false)
+
+    /// Original content of the order customer provided note.
+    ///
+    private let originalNote: String
+
+    /// Tracks if a network request is being performed.
+    ///
+    private let performingNetworkRequest: CurrentValueSubject<Bool, Never> = .init(false)
+
+    convenience init(order: Order) {
+        let note = order.customerNote ?? ""
+        self.init(originalNote: note)
+    }
+
+    /// Member wise initializer
+    ///
+    init(originalNote: String) {
+        self.originalNote = originalNote
+        self.newNote = originalNote
+        bindNavigationTrailingItemPublisher()
+    }
+
+    /// Update the note remotely and invoke a completion block when finished
+    ///
+    func updateNote(onFinish: @escaping () -> Void) {
+        // TODO: Fire network request
+        performingNetworkRequest.send(true)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3) { [weak self] in
+            self?.performingNetworkRequest.send(false)
+            onFinish()
```

Comment:
I wasn't very happy to add a completion block and break the reactive chain but sending this `onFinish` signal via a `@Published` variable seemed like an over-complication 😅